ICAFIX/hcp_fix_multi_run

```diff
@@ -379,116 +385,352 @@ log_Msg "PWD: $DIR"
+						# Use interpreted MATLAB or Octave
+						# ${hp} needs to be passed in as a string, to handle the hp=pd* case
+						matlab_code="${ML_PATHS} functionhighpassandvariancenormalize(${tr}, '${hp}', '${fmri}', '${Caret7_Command}', '', 1, ${ciftiwisharts}, '${icadimmode}');"
+						log_Msg "Run interpreted MATLAB/Octave with command..."
+						log_Msg "${FSL_FIXDIR}/call_matlab.sh -r ${matlab_code}"
+						${FSL_FIXDIR}/call_matlab.sh -r "${matlab_code}"   # Note: the semicolon within ${matlab_code} needs to be protected with quotation marks
+						;;
+					*)
+						# Unsupported MATLAB run mode
+						log_Err_Abort "Unsupported MATLAB run mode value (FSL_FIX_MATLAB_MODE) in $FSL_FIXDIR/settings.sh: ${FSL_FIX_MATLAB_MODE}"
+						;;
+				esac
+
+				log_Msg "Dims: $(cat ${fmri}_dims.txt)"
+
+				# Demean the movement regressors (in the 'fake-NIFTI' format returned by functionhighpassandvariancenormalize)
+				fslmaths ${fmri}_hp${hp}.ica/mc/prefiltered_func_data_mcf_conf -Tmean ${fmri}_hp${hp}.ica/mc/prefiltered_func_data_mcf_conf_mean
+				fslmaths ${fmri}_hp${hp}.ica/mc/prefiltered_func_data_mcf_conf -sub ${fmri}_hp${hp}.ica/mc/prefiltered_func_data_mcf_conf_mean ${fmri}_hp${hp}.ica/mc/prefiltered_func_data_mcf_conf
+				$FSLDIR/bin/imrm ${fmri}_hp${hp}.ica/mc/prefiltered_func_data_mcf_conf_mean
+
+				cd ${DIR}  # Return to directory where script was launched
+				
+				log_Msg "Bottom of loop through fmris: fmri: ${fmri}"
+			fi
+		done
 	fi
 
-	# Get Movement_Regressors.txt into the format expected by functionhighpassandvariancenormalize.m
-	mkdir -p ${fmri}_hp${hp}.ica/mc
-	if [ -f Movement_Regressors.txt ] ; then
-		log_Debug_Msg "About to create ${fmri}_hp${hp}.ica/mc/prefiltered_func_data_mcf.par file"
-		cat Movement_Regressors.txt | awk '{ print $4 " " $5 " " $6 " " $1 " " $2 " " $3}' > ${fmri}_hp${hp}.ica/mc/prefiltered_func_data_mcf.par
+	## ---------------------------------------------------------------------------
+	## Concatenate the individual runs and create necessary files
+	## ---------------------------------------------------------------------------
+
+	#Make Concatenated Folder
+	ConcatFolder=`dirname ${ConcatName}`
+	log_Msg "Making concatenated folder: ${ConcatFolder}"
+	if [ ! -e ${ConcatFolder} ] ; then  
+		mkdir ${ConcatFolder} 
 	else
-		log_Err_Abort "Movement_Regressors.txt not retrieved properly." 
+		/bin/rm -r ${ConcatFolder}
+		log_Warn "Previous ${ConcatFolder} directory removed"
+		mkdir ${ConcatFolder}
 	fi
 
-	#Demean movement regressors
-	demeanMovementRegressors Movement_Regressors.txt Movement_Regressors_demean.txt
-	MovementTXTMergeARRAY+=("$(pwd)/Movement_Regressors_demean.txt")
+	ConcatNameNoExt=$($FSLDIR/bin/remove_ext $ConcatName)  # No extension, but still includes the directory path
+
+	# Merge volumes from the individual runs
+	fslmerge -tr "${ConcatNameNoExt}_demean" "${NIFTIvolMergeARRAY[@]}" $tr
+	fslmerge -tr "${ConcatNameNoExt}_hp${hp}_vnts" "${NIFTIvolhpVNMergeARRAY[@]}" $tr
```

Comment:
With "hide whitespace", you can see these used to be `-volume-merge`s.  Was this intentional?  Perhaps to reduce memory footprint?

Reply:
Perhaps they were working from an old version of the code?  The commits are from a while ago...